lib/ansible/module_utils/lxd.py

```diff
@@ -121,6 +121,8 @@ def _send_request(self, method, url, body_json=None, ok_error_codes=None, timeou
             if resp_type == 'error':
                 if ok_error_codes is not None and resp_json['error_code'] in ok_error_codes:
                     return resp_json
+                if resp_json['error'] == "Certificate already in trust store":
```

Comment:
@june07 Thanks for your pull request.

At first, I thought I do not like depending on the particular error message and it would be better to call the `GET /1.0/certificates` or `GET /1.0/certificates/<fingerprint>` beforehand to check the certificate is in trust store.

However, since the LXD REST API does not have concept of transaction, after checking with one of the two APIs and before calling `POST /1.0/certificates`, there may be other requests to register the same certificate.
So I think checking the error of `POST /1.0/certificates` is the way to go.

I looked at the LXD code and confirmed the error message is the only way to check the error type.
https://github.com/lxc/lxd/blob/26858d333f13ac5e97bbc40a542374c7ec9156c3/lxd/certificates.go#L147

So after all, I agree to your pull request.